crates/settlement-clients/ethereum/src/conversion.rs

```diff
@@ -1,12 +1,325 @@
+        ]
+    )]
+    fn slice_slice_u8_to_vec_u256_works(#[case] slices: &[[u8; 32]], #[case] expected: Vec<U256>) {
+        match slice_slice_u8_to_vec_u256(slices) {
+            Ok(response) => {
+                assert_eq!(response, expected);
+            }
+            Err(e) => {
+                panic!("{}", e);
+            }
+        }
+    }
+
+    #[rstest]
+    #[case::empty(&[], "0".repeat(64))]
+    #[case::typical(&[0xFF,0xFF,0xFF,0xFF], format!("{}{}", "ff".repeat(4), "0".repeat(56)))]
+    #[case::big(&[0xFF; 32], format!("{}", "ff".repeat(32)))]
+    fn to_hex_string_works(#[case] slice: &[u8], #[case] expected: String) {
+        let result = to_padded_hex(slice);
+        assert_eq!(result, expected);
+        assert!(expected.len() == 64);
+    }
+
+    #[rstest]
+    #[should_panic(expected = "Slice length must not exceed 32")]
+    #[case::exceeding(&[0xFF; 40], format!("{}", "ff".repeat(32)))]
+    fn to_hex_string_panics(#[case] slice: &[u8], #[case] expected: String) {
+        let _ = to_padded_hex(slice);
+        assert!(expected.len() == 64);
+    }
+
+    #[rstest]
+    #[case::typical([
+        1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
+        31, 32, 33, 34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48,
+    ],
+    "0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f202122232425262728292a2b2c2d2e2f3000000000000000000000000000000000")]
+    #[case::single_value(
+        [0xFF;48],
+        format!("{}{}","ff".repeat(48), "00".repeat(16))
+    )]
+    fn u8_48_to_hex_string_works(#[case] slice: [u8; 48], #[case] expected: String) {
+        let result = u8_48_to_hex_string(slice);
+        assert_eq!(result, expected);
+    }
+
+    #[rstest]
+    #[case::typical(
+        vec![
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32],
+            [32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1],
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 0, 0]
+        ],
+        "0102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f20201f1e1d1c1b1a191817161514131211100f0e0d0c0b0a0908070605040302010102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e0000"
+    )]
+    #[case::single_value(
+        vec![
+            [0xFF;32],
+            [0xF5;32],
+        ],
+        format!("{}{}", "ff".repeat(32), "f5".repeat(32))
+    )]
+    fn vec_u8_32_to_hex_string_works(#[case] slice: Vec<[u8; 32]>, #[case] expected: String) {
+        let result = vec_u8_32_to_hex_string(slice);
+        assert_eq!(result, expected);
+    }
+
+    #[rstest]
+    #[case("20462788")]
```

Comment:
let's add a small desc saying that we're matching commitments with mainnet over here

Reply:
Valid, Added.

Reply:
can we also add a blobscan/etherscan link with actual blob 🙏

Reply:
Alright, Added.